src/node_url_pattern.cc

```diff
@@ -165,6 +165,10 @@ void URLPattern::New(const FunctionCallbackInfo<Value>& args) {
     input = input_buffer.ToString();
   } else if (args[0]->IsObject()) {
     init = URLPatternInit::FromJsObject(env, args[0].As<Object>());
+    // If init does not have a value here, the implication is that an
+    // error was thrown. Let's allow that to be handled now by returning
+    // early. If we don't, the error thrown will be swallowed.
+    if (!init.has_value()) return;
```

Comment:
```suggestion
    if (!init) return;
```

Reply:
Beyond being less verbose is there a particular reason to prefer to `(!init)` or `(!init.has_value())` ? If not, I think I'd rather keep these as is.

Reply:
In all other places I used the bool operator. For consistency, I recommend doing the same thing.